app/views/transaction.py

```diff
@@ -0,0 +1,766 @@
+from app.schemas.transaction import account_transaction_details_schema, accounts_transaction_details_schema, transaction_type_schema, transactions_type_schema, fund_transfer_schema, funds_transfer_schema
+from app.common.response_genarator import ResponseGenerator
+from http import HTTPStatus
+from sqlalchemy.orm.exc import NoResultFound
+from sqlalchemy import desc
+
+
+class AccountTransactionDetailsResource(Resource):
+    def post(self):
+        """
+              This is POST API
+              Create a new account transaction
+             parameters:
+                 transaction_amount: Integer
+                 transaction_date: DateTime
+                 bank_account_id: Integer
+                 transaction_type_id: Integer
+                 fund_transfer_id: Integer
+             responses:
+                 404:
+                     description: Account transaction details does not exist
+                 200:
+                     description: Account transaction details record inserted successfully
+                     schema:
+                         AccountTransactionDetailsSchema
+         """
+        # retrieve body data from input JSON
+
+        data = request.get_json()
+        errors = account_transaction_details_schema.validate(data, partial=True)
+        if errors:
+            logger.error("Missing or sending incorrect data to create an activity {}".format(errors))
+            response = ResponseGenerator(data={},
+                                         message=errors,
+                                         success=False,
+                                         status=HTTPStatus.BAD_REQUEST)
+            return response.error_response()
+
+        try:
+            # get bank account details
+            bank_account = BankAccount.query.filter(BankAccount.id == data['bank_account_id']).first()
+            if not bank_account:
+                raise BankAccountObjectNotFound("Invalid Bank Account")
+
+            # get transaction type details
+            transactions_type = TransactionType.query.filter(TransactionType.id == data['transaction_type_id']).first()
+            if not transactions_type:
+                raise TransactionTypeObjectNotFound("Invalid Transaction Type")
+
+            # for credit
+            if transactions_type.transaction_type.lower() == 'credit':
+                bank_account.account_balance += data['transaction_amount']
+                db.session.add(bank_account)
+                transaction_status = "success"
+            # for debit
+            elif transactions_type.transaction_type.lower() == 'debit':
+                if bank_account.account_balance - 1000 - data['transaction_amount'] > 0:
+                    bank_account.account_balance -= data['transaction_amount']
+                    db.session.add(bank_account)
+                    transaction_status = "success"
+                else:
```

Comment:
We should raise some exception if low balance.

Reply:
Made the changes as suggested.